otherlibs/dune-rpc/dune_rpc.mli

```diff
@@ -82,25 +82,56 @@ module V1 : sig
       }
   end
 
-  module Error : sig
+  module Diagnostic : sig
+    type severity =
+      | Error
+      | Warning
+
+    module Promotion : sig
+      type t =
+        { in_build : string
+        ; in_source : string
+        }
+    end
+
     type t =
       { target : string option
       ; message : string
       ; loc : Loc.t option
+      ; severity : severity option
+      ; promotion : Promotion.t option
       }
+
+    module Event : sig
+      type nonrec t =
+        | Add of t
+        | Remove of t
+    end
   end
 
-  module Promotion : sig
+  module Build : sig
+    type t
```

Comment:
Is the intention to eventually add functions consuming a `Build.t` to this module? At the moment, the optional `build_event` and `build_progress` arguments to `create` can't actually do anything other than ignore that argument.

Reply:
Yes, I'm thinking of adding functions such as:

```ocaml
val targets : Build.t -> target list
```